src/serialization/interval_log/mod.rs

```diff
@@ -0,0 +1,667 @@
+        let mut internal_writer = InternalLogWriter {
+            writer,
+            serializer,
+            text_buf: String::new(),
+            serialize_buf: Vec::new(),
+            max_value_divisor: self.max_value_divisor,
+        };
+
+        for c in self.comments.iter() {
+            internal_writer.write_comment(&c)?;
+        }
+
+        if let Some(st) = self.start_time {
+            internal_writer.write_fmt(format_args!(
+                "#[StartTime: {:.3} (seconds since epoch)]\n",
+                st
+            ))?;
+        }
+
+        if let Some(bt) = self.base_time {
+            internal_writer.write_fmt(format_args!(
+                "#[BaseTime: {:.3} (seconds since epoch)]\n",
+                bt
+            ))?;
+        }
+
+        // The Java impl doesn't write a comment for this but it's confusing to silently modify the
+        // max value without leaving a trace
+        if self.max_value_divisor != 1.0_f64 {
+            internal_writer.write_fmt(format_args!(
+                "#[MaxValueDivisor: {:.3}]\n",
+                self.max_value_divisor
+            ))?;
+        }
+
+        Ok(IntervalLogWriter { internal_writer })
+    }
+}
+
+/// Writes interval histograms in an interval log.
+///
+/// This isn't created directly; start with an `IntervalLogWriterBuilder`. Once you've written the
+/// headers and ended up with an `IntervalLogWriter`, typical usage would be to write a histogram
+/// at regular intervals (e.g. once a second).
+///
+/// ```
+/// use hdrsample::serialization;
+/// use hdrsample::serialization::interval_log;
+///
+/// let mut buf = Vec::new();
+/// let mut serializer = serialization::V2Serializer::new();
+///
+/// // create a writer via a builder
+/// let mut writer = interval_log::IntervalLogWriterBuilder::new()
+///     .build_with(&mut buf, &mut serializer)
+///     .unwrap();
+///
+/// writer.write_comment("Comment 2").unwrap();
+///
+/// // .. write some intervals
+/// ```
+pub struct IntervalLogWriter<'a, 'b, W: 'a + io::Write, S: 'b + Serializer> {
+    internal_writer: InternalLogWriter<'a, 'b, W, S>,
+}
+
+impl<'a, 'b, W: 'a + io::Write, S: 'b + Serializer> IntervalLogWriter<'a, 'b, W, S> {
+    /// Add a comment line.
+    pub fn write_comment(&mut self, s: &str) -> io::Result<()> {
+        self.internal_writer.write_comment(s)
+    }
+
+    /// Write an interval histogram.
+    ///
+    /// `start_timestamp` is the time since the epoch in seconds that measurements started being
+    /// recorded in this interval. If you're using a StartTime or BaseTime offset, you should
+    /// instead use a delta since that time. See the discussion about timestamps in the module-level
+    /// documentation.
+    ///
+    /// `duration` is the duration of the interval in seconds.
+    ///
+    /// `tag` is an optional tag for this histogram.
+    pub fn write_histogram<T: Counter>(
+        &mut self,
+        h: &Histogram<T>,
+        start_timestamp: f64,
```

Comment:
I wonder if this should maybe be a `Duration` as well? 🤔 Conceivably, that's how it would be used: a `Duration` since either `UNIX_EPOCH` or whatever the chosen `StartTime`/`BaseTime` was.

Reply:
Oh, yeah, that's a good idea!